axelrod/strategies/lookerup.py

```diff
@@ -0,0 +1,150 @@
+    opponents first n moves, self's last m moves, and opponents last m moves,
+    all as strings. The values are the moves to play on this round.  For
+    example, if
+
+    - the opponent started by playing C  
+    - my last move was a C the opponents
+    - last move was a D
+
+    the corresponding key would be
+
+        ('C', 'C', 'D')
+
+    and the value would contain the move to play on this turn.
+
+    Some well-known strategies can be expressed as special cases; for example
+    Cooperator is given by the dict:
+
+        {('', '', '') : 'C'}
+
+    where m and n are both zero. Tit For Tat is given by:
+
+    { 
+        ('', 'C', 'D') : 'D',         
+        ('', 'D', 'D') : 'D',         
+        ('', 'C', 'C') : 'C',
+        ('', 'D', 'C') : 'C',     
+    }
+
+    where m=1 and n=0.
+
+    Lookup tables where the move depends on the opponent's first moves (as
+    opposed to most recent moves) will have a non-empty first string in the
+    tuple. For example, this fragment of a dict:
+
+    {          
+        ...         
+        ('C', 'C', 'C') : 'C'.
+        ('D', 'C', 'C') : 'D',
+        ...            
+    }
+
+    states that if self and opponent both cooperated on the previous turn, we
+    should cooperate this turn unless the opponent started by defecting, in
+    which case we should defect.
+
+    """
+    
+
+    name = 'LookerUp'
+    classifier = {
+        'memory_depth': float('inf'), 
+        'stochastic': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self, lookup_table=None):
+        """
+        If no lookup table is provided to the constructor, then use the TFT one
+        """
+        Player.__init__(self)
+
+        if not lookup_table:
+            lookup_table = { 
+            ('', 'C', 'D') : 'D',         
+            ('', 'D', 'D') : 'D',         
+            ('', 'C', 'C') : 'C',
+            ('', 'D', 'C') : 'C',     
+        }
+
+        self.lookup_table = lookup_table
+
+
+        # rather than pass number of previous rounds to consider in as a separate variable, figure it out
+        # the number of rounds is the length of the second element of any given key in the dict 
+        self.plays = len(list(self.lookup_table.keys())[0][1]) 
+
+        # the number of opponent starting moves is the lgnth of the first element of any given key in the dict
+        self.opponent_start_plays = len(list(self.lookup_table.keys())[0][0])
+
+        if self.opponent_start_plays == 0:
+            self.classifier['memory_depth'] = self.plays
+
+        self.init_args = (lookup_table,)
+
+    def strategy(self, opponent):
+        """
+        If there isn't enough history to lookup a move from the table, cooperate
+        """
+        if len(self.history) < max([self.plays,self.opponent_start_plays]) :
+            return 'C'
+            
+        else:
```

Comment:
No need for `else` as you have a `return` (this is stylistic so feel free to ignore).